lightning/src/util/test_utils.rs

```diff
@@ -71,6 +72,20 @@ impl chaininterface::FeeEstimator for TestFeeEstimator {
 	}
 }
 
+pub struct TestRouter {}
+
+impl Router for TestRouter {
+	fn find_route(
+		&self, _payer: &PublicKey, _params: &RouteParameters, _first_hops: Option<&[&channelmanager::ChannelDetails]>,
+		_inflight_htlcs: InFlightHtlcs
+	) -> Result<Route, msgs::LightningError> {
+		Err(msgs::LightningError {
```

Comment:
Running into some lifetime woes. Is it fine if I make the `Node` `TestLogger` into an `Arc`? I think otherwise we'll need to add another `create_chanmon_cfg_cfgs` ahead to make the lifetimes work

Reply:
https://github.com/valentinewallace/rust-lightning/tree/2022-10-chanman-router-param-wip